scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -543,7 +544,15 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
                 newRegions
             } else sepRegions
           (nextRegions, currRef)
-        } else (sepRegions, currRef)
+        } else if (dialect.allowSignificantIndentation && curr.is[KwFor]) {
```

Comment:
is `KwFor` the only such case?

Reply:
sepRegions is used to simplify handling of whitespaces, is some of them they are automatically omitted so that the parser doesn't have to handle it.

I think KW is the only case like that, can't think of another really.